Objects/typeobject.c

```diff
@@ -2994,7 +3010,7 @@ _PyType_Lookup(PyTypeObject *type, PyObject *name)
         assert(PyType_Check(base));
         dict = ((PyTypeObject *)base)->tp_dict;
         assert(dict && PyDict_Check(dict));
-        res = PyDict_GetItem(dict, name);
+        res = _PyDict_GetItem_KnownHash(dict, name, hash);
```

Comment:
`PyDict_GetItem()` always clears errors. Call `PyErr_Clear()` if `_PyDict_GetItem_KnownHash()` returns NULL.

Reply:
Already changed.

Reply:
I don't see well the difference between PyDict_GetItem() and _PyDict_GetItem_KnownHash() in term of performance. Maybe the difference is that PyDict_GetItem() calls PyErr_Fetch/PyErr_Restore. In that case, PyDict_GetItemWithError() would have the same speed, no?

Reply:
Using `PyDict_GetItemWithError()` adds only a half of the speed up.

The difference is:
1. PyDict_GetItem() calls PyThreadState_GET/PyErr_Fetch/PyErr_Restore.
2. PyDict_GetItem() checks for str and reads a hash every time. I don't understand well why this affects performance.

Reply:
KnownHash is extremely short in comparison and probably gets inlined and streamlined with LTO. Substantially less branching.